### PR TITLE
ARCH-687: feat: use configured logging service

### PR DESCRIPTION
**Pull request by robrap**
The loggingService is now configured from the app, which enables us to
use a shared library.

BREAKING CHANGE: The configuration of the authClient now requires you
to pass it a loggingService.

ARCH-687

### CODE REVIEW

File: src/AuthenticatedAPIClient/tests/AuthenticatedAPIClient.test.jsx [hunk 1]
```diff
@@ -141,14 +143,6 @@ describe('getAuthenticatedAPIClient', () => {
   });
 });
 
-describe('logAPIErrorResponse', () => {
```
Comment (robrap) [inline src/AuthenticatedAPIClient/tests/AuthenticatedAPIClient.test.jsx]: Reviewers: FYI: Not sure what this was going to test, but it is a dup of the test above it and doesn't test api responses.  Note that I will be working in that area, so maybe I'll add a test then.

File: src/logging.js [hunk 2]
```diff
@@ -1,27 +1,42 @@
-function logInfo(message) {
-  if (typeof newrelic !== 'undefined') {
-    newrelic.addPageAction('INFO', { message }); // eslint-disable-line no-undef
+let loggingService = null;
```
Comment (davidjoy) [inline src/logging.js]: This file is interesting, in that it seems like these checks would be useful to any library configuring a logging service... does this belong in frontend-auth?

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
+frontend-auth
```
Comment (albemarle) [inline README.rst]: Nice doc.

**Comment by robrap** [comment]
@albemarle @davidjoy: FYI: I have moved much of this to frontend-logging, and started a new PR there: https://github.com/edx/frontend-logging/pull/6
I will update this PR to reflect the move.

File: src/logging.js (hunk 2 above)
Comment (robrap) [reply src/logging.js]: Moved to https://github.com/edx/frontend-logging/pull/6

File: package.json [hunk 4]
```diff
@@ -73,6 +73,7 @@
     "redux": "^4.0.0"
   },
   "dependencies": {
+    "@edx/frontend-logging": "file:../frontend-logging",
```
Comment (robrap) [inline package.json]: Reviewers: I will update this before landing the PR.

**Pull request merged**